www/js/diary/infinite_scroll_list.js

```diff
@@ -236,7 +237,8 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
                 Logger.displayError("multiple filters not supported!", undefined);
             } else {
                 // console.log("Trip n before: "+$scope.data.displayTrips.length);
-                $scope.data.displayTrips = $scope.data.allTrips.filter(f.filter);
+                $scope.data.displayTrips = $scope.data.allTrips.filter(
```

Comment:
This is quite an edge case, but: what if the user goes to All Unlabeled, labels a trip that is not in To Label (triggering the delayed disappearance) and then switches to To Label? We want the trip to disappear immediately, given that it should not be on To Label at all. Will this keep it around?

Reply:
Good catch! This will keep it around.

I am not sure I agree with "we want the trip to disappear immediately" though. They did label it very very recently and maybe they still want to change it. In other words, the simple rule for labeling would be "we will give you some time to fix your changes" and showing it in "To Label" will keep that rule.

If that is too confusing, the fix is fairly straightforward - any time we change the selection, we reset the `timeoutPromise` and the `trip.waitingForMod` for all trips.

Reply:
I think it will be confusing to break the promise of what the filters do in that way (even though we are already breaking that promise by having labeled trips stick around). Ideal would be to have the trip disappear immediately if you switch to To Label, but then if you switch back to All Unlabeled have it still be there for the rest of the minute. That sounds more complicated to implement though.

Reply:
That is way too complicated to implement.

But going back to the promise of what the filters do, my argument is that with this change, we are fundamentally altering the promise of what those filters do (i.e. "even though we are already breaking that promise by having labeled trips stick around").

The new promise is: the "to label" screen will have the trips that you need to label, and any trips that you have labeled recently, giving you a chance to correct them.

This implementation is completely consistent with that promise. I would argue that it is actually *more* confusing to have the behavior be different if the labeling happens directly from the "To Label" screen versus the "All Trips" -> "To Label" screen